File: java/utility/ResultHeaders.java
package voxgig.bluefinshieldconexmgmtsdk.utility;

import java.util.LinkedHashMap;
import java.util.Map;

import voxgig.bluefinshieldconexmgmtsdk.core.Context;
import voxgig.bluefinshieldconexmgmtsdk.core.Helpers;
import voxgig.bluefinshieldconexmgmtsdk.core.Response;
import voxgig.bluefinshieldconexmgmtsdk.core.Result;

final class ResultHeaders {

  private ResultHeaders() {}

  static Result resultHeaders(Context ctx) {
    Response response = ctx.response;
    Result result = ctx.result;

    if (result != null) {
      if (response != null && response.headers != null) {
        Map<String, Object> hm = Helpers.toMapAny(response.headers);
        if (hm != null) {
          result.headers = hm;
        }
        else {
          result.headers = new LinkedHashMap<>();
        }
      }
      else {
        result.headers = new LinkedHashMap<>();
      }
    }

    return result;
  }
}

File: .sdk/tm/php/utility/ResultHeaders.php
<?php
declare(strict_types=1);

// BluefinShieldconexMgmt SDK utility: result_headers

class BluefinShieldconexMgmtResultHeaders
{
    public static function call(BluefinShieldconexMgmtContext $ctx): ?BluefinShieldconexMgmtResult
    {
        $response = $ctx->response;
        $result = $ctx->result;
        if ($result) {
            if ($response && is_array($response->headers)) {
                $result->headers = $response->headers;
            } else {
                $result->headers = [];
            }
        }
        return $result;
    }
}
